notebooks/textbook/Recursive_Quantum_Approximate_Optimization_Algorithm.ipynb

```diff
@@ -0,0 +1,579 @@
iDZlPT9pj04jdyPr6DYrDNk6uqDm/ED6Y8v4/OBGdAauR6ymvpF3rfs8n/o3EST+X4ZGKoBKdmEzzpdgHyuRPoCGCdxOfA69xwcAfKJ/7i9H3I6TUE8HnjZaSWW5fAI804+hhXrJQICAhAWFgYlJSUMGjQIO3bsgJWVlcRP/otj/vz5OHToEAICAjBu3DgAQE5ODjZs2ICRI0eiQYMGIpZQMCZMmICGDRtiyJAh6N69Oy5fvox69eqVeE9eXh4mTpyIgwcPYtWqVfD09BS79G9FwWazsWDBAowaNQrPnz/nn5v86tUrnDt3Dnv37q1272v9+vUxd+5czJ07F8+ePUNgYCCOHDmCHTt2oEmTJnBxcYGLiwsMDQ1/6QhCkUYhN/E1sl6FQs1qLGqZOQIAlE16IHHfFKT+cwBaruuLbJ/DI3ide16h1IQMkouFhQW8vLywfPly9O7dGx07dixwncPhwMXFBVpaWlWaeUFKSgrDhg3D0KFDcfnyZaxatQp2dnZo27YtvLy8hOLA5XK5OHXqFFavXs0/4+nq1avo3bu3ROhLYWFrawsdHR34+fnxncR+fn7Q1NREv379RCydcGGz2WjQoAEaNGiAbt26FbqekZFRIAr5tzP56tWrJUYh//2pV69ehd4hdXV1rFixArNnz8aOHTuwadMmnP6mBQW91kXWy+h/hrLSuXNnrF69GvPnz0fnzp1hb28vapEkGjabjebNm6N58+Zwc3MD8Cs12P379/mRxgsXLkROTg7k5ORQf9I+cBXK5wQuDS6PcOHhO/y8vY/vEE5MTATwy7Y2adIEJiYmcHd350cGN27cWGycp5XBlClT4OfnhwsXLojFZlM/Pz9MmjQJQ4YMweHDhyvk4I2IiICDg4MQpWNgEB316tXD1atXYWFhAUdHR1y7dq3Y38f69euxa9cu+Pn5oU+fPlUsKQODYCgoKPCdwQCQnJyMkJAQBAcH4/Tp09i8eTNkZGRgaWnJL9e2bdtCa14lrRUz8wDJp6TvV6XDQGj0mweW1P/GaUrNuyBx/1SkPTgNDYe5Rd7HfL8MDNUHvo6Q0LWg6h96I2TeJqXj7ruUUg+dzol/gazoMNTuOUGgerk8woO4n5ix2Aeqqqo4cuQIkpKScOjQIVhbW1c7h9ufNGvWDI6Ojli7di24XC6AX6mXk5KS4OHhIWLpyoatrS1CQ0Px8eNHmJubIzo6utiyP3/+hL29PT9i3MvLS6IcHs7OztDT08OaNWv4f1u7di10dHTg6uoqQskqn1atWsHX1xfx8fG4efMmLCwssG7dOjRq1AgdbfqXqCOyXocBLDZU2vxvksySloVyaxvkfooGJ+1rkfdxeYS771LwLjm9UvrEIP4sWbIEpqamcHFxQWZmZoFrq1atwsOHDxEYGFjk+e6VDYvFgoODA+7fv4+bN2+iVq1aGDx4MExMTBAQEID8/Pwy15mXlwd/f380b94czs7O0NbWxp07d3D37t1KiVQWd6SlpTF27FgcPXoUGRkZyMnJQUBAAEaPHl3jIpOUlZXRsmVL9O/fH7NmzcL27dsRFBSE6OhoZGdnIyEhAXfu3MGBAwcwduxY6Orq4s2bN9ixYwdcXV1haWkJbW1tKCsrw8TEhF/Ptm3bcOXKFbx69QrZ2dkCy6OmpoaFCxciJOol5PVNQayih9eM/mcoD3PnzoW9vT1GjhyJ+Ph4UYtT7dDQ0ICDgwN8fHxw584d/Pz5ExEREVixei248mqV2vb3PClcunoDioqKGDNmDI4ePYp///0XmZmZePXqFU6dOoVly5Zh8ODBaN68ebV2EANA69at0blzZ2zfvl2kchAR1qxZgwkTJmDSpEk4cuRIhexsSkoKYmJiYGbGLAAzVB+aNWuGCxcuICwsDG5ubijqFLsTJ07Aw8MDixYt4gcDMDBIAnXr1oWzszP8/f0RHx+PV69eYcOGDVBRUYGPjw86duwITU1NDBkyBHv37kVsbGypa8XMPECyKe37lW/QvICDGABk6tSHrIYu8lMSiq2X+X4ZGKoH1cEGME7iMvI77VhJEI+L78G7ody6F2Tr6gtcNwuE6dtOISgoCMOHDy93OitJxNPTE+/evcPp06fB4XCwdu1aDBkyBEZGRqIWrcy0bdsWDx48gKKiIiwsLBAaWvgss48fP6JLly6IiorC9evXMWLECBFIWjFkZGQwb948HD9+HO/fv0dcXByOHDmCOXPmFEgbXp2RkpJCz549cfDgQSQlJeHo0aPg6JmDeNxi78lLioFMnfpgyxVMFSGrbcS/Xmx7/5+akKFmIiMjgyNHjuDTp08F0vBHRERg5cqVWLRoEczNzUUo4S9ncc+ePXHr1i2Eh4fD0NAQI0eOhJGREXbv3o2cnJxS68jOzsa2bdvQuHFjuLm5wdjYGJGRkbh+/Tq6du1aBb0QX9zc3JCRkYGTJ0/izJkz+PHjB7Po9he/o5C7du2K0aNHY8WKFQgMDER4eDiSkpKQnp6OZ8+e4fz58/D29oaVlRU4HA6uXbuGefPmoW/fvmjRogUUFRVRv359dOnSBaNGjcLy5cv59Xz58qXIxdBzz1NKHCMy+p+hPLDZbBw6dAjKysoYNmxYgUh5BuEjKyuLjh07wt5pdJG7wIUJi8XChVvhfH3k7OyMVq1a1ZhxdFFMnToVd+7cwYsXL0TSPo/Hw9y5c+Hp6YmlS5di+/btFd6sHRkZCQCFssAwMEg6Xbp0weHDhxEYGIglS5YUuHb37l2MHDkSrq6uWLFihYgkZGCoOCwWC82aNcO0adNw8eJFfP/+HXfv3sW0adPw8eNHTJ48GYaGhug1ZSVYxCu2HmYeINkI4gv4GyICNysVbMWSz0Fmvl8GBsmnNB0hCTaASTddRm6/Ti41ijjjyVVw0r6invOqMtVNYCEiIaMi4kks7dq1Q69eveDj4wMOh4PY2FicOXNG1GKVGz09PYSFhcHR0RE2NjY4dOgQ3+H9+vVrODg4QEZGBuHh4
```

Comment:
We should run against a real device. I can help you identify a path to get credits if necessary. I will follow-up via email.